### ISSUE TITLE
[win] i-Builds repeatedly DNF I20241211-0310

### ISSUE THREAD

**Issue by jukzi** [eclipse-platform/eclipse.platform.releng.aggregator#2651]
https://ci.eclipse.org/releng/job/AutomatedTests/job/ep435I-unit-win32-x86_64-java17/
![image](https://github.com/user-attachments/assets/0a23d955-005c-47a3-9773-0eb9a0a58153)

**Comment by sratz** [comment]
I opened #2652. Maybe with that all tests can run and the log file gives us some indication for which test suites `msedgewebview2.exe` were lingering after platform shutdown to find some pattern.

**Comment by fedejeanne** [comment]
> Cannot reproduce locally:
> 
> ```
> java -jar plugins\org.eclipse.equinox.launcher_1.6.900.v20240613-2009.jar -application org.eclipse.test.uitestapplication -data .ws -testPluginName org.eclipse.update.configurator.tests -className org.eclipse.update.configurator.tests.AutomatedConfiguratorSuite -os win32 -ws win32 -arch x86_64 -consolelog
> rmdir /s /q .ws
> ```
> 
> is able to delete the whole workspace without problem :(

Can you try that in a loop, say 100 times?

Or can you tell me where exactly did you run the command so I can test it too?

**Comment by sratz** [comment]
Looks like this helper. All tests have run again and the log says that some `msedgewebview2.exe` instances were indeed killed.

This is something we need to handle more nicely in the test infrastructure than just killing processes.

But in general this is something that we will have to deal with with Edge in some way or the other due to WebView2's multiprocess / asynchronous architecture.

**Comment by fedejeanne** [comment]
> Looks like this helped. All tests have run again and the log says that some `msedgewebview2.exe` instances were indeed killed.

Awesome, thank you so much for the fix!
 
> This is something we need to handle more nicely in the test infrastructure than just killing processes.
> 
> But in general this is something that we will have to deal with with Edge in some way or the other due to WebView2's multiprocess / asynchronous architecture.

Agreed. We should revisit this issue early next year.

**Comment by HeikoKlare** [comment]
Thank you all for having a look and thank you @sratz for finding a solution so fast (even if it may not be the final solution)!
Also great to see that all downstream tests of the I-Build seem to run fine in the Jenkins infrastructure with Edge being used as default.

**Comment by merks** [comment]
Indeed. Thanks to all of you. Also @jukzi and @akurtakov.  So many people involved to keep things running smoothly. Much of it thankless work. But essential to running a good ship.
